x-pack/plugins/apm/public/components/alerting/ui_components/chart_preview/chart_preview_helper.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiLoadingChart } from '@elastic/eui';
+import { EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { niceTimeFormatter } from '@elastic/charts';
+import { min as getMin, max as getMax } from 'lodash';
+import { Coordinate } from '../../../../../typings/timeseries';
+
+export const TIME_LABELS = {
+  s: i18n.translate('xpack.apm.alerts.timeLabels.seconds', {
+    defaultMessage: 'seconds',
+  }),
+  m: i18n.translate('xpack.apm.alerts.timeLabels.minutes', {
+    defaultMessage: 'minutes',
+  }),
+  h: i18n.translate('xpack.apm.alerts.timeLabels.hours', {
+    defaultMessage: 'hours',
+  }),
+  d: i18n.translate('xpack.apm.alerts.timeLabels.days', {
+    defaultMessage: 'days',
+  }),
+};
+
+export const useDateFormatter = (xMin?: number, xMax?: number) => {
+  const dateFormatter = useMemo(() => {
+    if (typeof xMin === 'number' && typeof xMax === 'number') {
+      return niceTimeFormatter([xMin, xMax]);
+    } else {
+      return (value: number) => `${value}`;
+    }
+  }, [xMin, xMax]);
+  return dateFormatter;
+};
+
+export const getDomain = (
+  series: Array<{ name?: string; data: Coordinate[] }>
+) => {
+  let min: number | null = null;
+  let max: number | null = null;
+  const valuesByTimestamp = series.reduce<{ [timestamp: number]: number[] }>(
+    (acc, serie) => {
+      serie.data.forEach((point) => {
+        const valuesForTimestamp = acc[point.x] || [];
+        acc[point.x] = [...valuesForTimestamp, point.y ?? 0];
+      });
+      return acc;
+    },
+    {}
+  );
+  const pointValues = Object.values(valuesByTimestamp);
+  pointValues.forEach((results) => {
+    const maxResult = getMax(results);
+    const minResult = getMin(results);
+    if (maxResult && (!max || maxResult > max)) {
+      max = maxResult;
+    }
+    if (minResult && (!min || minResult < min)) {
+      min = minResult;
+    }
+  });
+  const timestampValues = Object.keys(valuesByTimestamp).map(Number);
+  const minTimestamp = getMin(timestampValues) || 0;
+  const maxTimestamp = getMax(timestampValues) || 0;
+  return {
+    yMin: min || 0,
+    yMax: max || 0,
+    xMin: minTimestamp,
+    xMax: maxTimestamp,
+  };
+};
```

Comment:
I don't quite understand what this code does. Simply reading the name and seeing the usage I'd expect this to suffice:

```suggestion
const xValues = series.flatMap(item => item.data.map(d => d.x))
const yValues = series.flatMap(item => item.data.map(d => d.y))
return {
  xMax: Math.max(...xValues),
  xMin: Math.min(...xValues),
  yMax: Math.max(...yValues),
  yMin: Math.min(...yValues)
};
```

I'm probably missing something.

Reply:
It can be simplified as you suggested. I will update the code.

Reply:
updated